omx/gstomxh264enc.c

```diff
@@ -577,6 +744,20 @@ gst_omx_h264_enc_handle_output_frame (GstOMXVideoEnc * enc, GstOMXPort * port,
     self->headers = NULL;
   }
 
+  if (self->idr_period) {
```

Comment:
Does the encoder not handle this on its own?

Reply:
No, this logic is part of a property to force an IDR frame which is something we need to set on the component configuration.